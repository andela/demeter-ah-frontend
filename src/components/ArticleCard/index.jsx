import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';
import { ImageStyle, authorImage } from '../../utils';
import BookmarkIcon from '../../assets/svgs/bookmarkfill';
import BookmarkEmpty from '../../assets/svgs/bookmark';
import Rating from '../Rating';

const ArticleCard = ({
  image,
  category,
  autImage,
  authorname,
  title,
  description,
  bookmark = false,
  emptyBookmark = false,
  onBookmark = () => {},
  slug,
  readTime,
  rating,
  createdAt,
  addBookmark,
  isBookmarked
}) => (
  <Fragment>
    <div className="md:flex md:flex-row flex-row w-full max-w-5xl m-auto text-left justify-start p-2
    bg-white rounded-sm border-b-2 border-gray-350 mb-3 md:min-w-84"
    >
      <div className="m-4 min-w-full ml-0 min-w-108 min-h-46 rounded shadow-lg md:min-w-min lg:min-w-84" style={ImageStyle(image || '/placeholder.png')}>
        <div className="m-5 flex flex-col justify-between h-82">
          <p className="text-xl font-medium text-white lg:mt-0 mt-4">
            {category}
          </p>
          <div className="flex justify-between mt-16">
            <div>
              <Link to={`/profile/${authorname || 'Paul Igabu'}/articles`}>
                <div style={authorImage(autImage || '/dummy.png')} className="h-16 w-16" />
                <p className="text-base text-left font-medium text-white">{authorname || 'Paul Igabu'}</p>
              </Link>
            </div>
            <span className="h-8" style={{ width: 'fit-content', alignSelf: 'flex-end', cursor: 'pointer' }}>
              {bookmark ? <BookmarkIcon onClick={onBookmark} slug={slug} /> : ''}
            </span>
            {emptyBookmark && isBookmarked
              ? (
                <span className="h-8 w-8" style={{ width: 'fit-content', alignSelf: 'flex-end', cursor: 'pointer' }}>
                  <BookmarkIcon onClick={addBookmark} slug={slug} />
                </span>
              ) : '' }
            {emptyBookmark && !isBookmarked
              ? (
                <span className="h-8 w-8" style={{ width: 'fit-content', alignSelf: 'flex-end', cursor: 'pointer' }}>
                  <BookmarkEmpty onClick={addBookmark} slug={slug} />
                </span>
              ) : '' }
          </div>
        </div>
      </div>
      <div className="ml-0 md:ml-8 flex-grow justify-between flex flex-col max-h-48">
        <div className="flex flex-col">
          <Link to={`/articles/${slug}`}>
            <h2 className="text-1.5xl lg:text-2xl capitalize">{title || 'Title of this amazing article'}</h2>
          </Link>
          <p className="text-xs">{description || 'Lorem ipsum dolor sit amet consectetur adipisicing elit. mus nihil eum iste! Dolorum, voluptatum officiis.'}</p>
        </div>
        <div className="flex justify-between mt-5">
          <div className="flex w-2/3 justify-between text-sm">
            <p className="pr-10 text-sm xl:text-base">{readTime || '6 min'}</p>
            <p className="font-semibold text-sm xl:text-base">{createdAt || 'July 12 2019'}</p>
          </div>
          <div className="flex w-1/3 justify-end items-center">
            <Rating rating={rating} />
          </div>
        </div>
      </div>
    </div>
  </Fragment>
);

export default ArticleCard;
