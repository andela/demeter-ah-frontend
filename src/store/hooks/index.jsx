import { useEffect } from 'react';
import { getUser } from '../actions/signup';

export const useSetUser = ({ dispatch }) => {
  useEffect(() => {
    dispatch(getUser());
  }, []);
  return [];
};
