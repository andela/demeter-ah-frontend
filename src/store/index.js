import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import combineReducers from './reducers';
import dummyAction from './actions/Signup';

const middlewares = applyMiddleware(thunk);

const store = createStore(
  combineReducers,
  composeWithDevTools(middlewares),
);

store.dispatch(dummyAction);

export default store;
