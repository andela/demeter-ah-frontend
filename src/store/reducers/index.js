import { combineReducers } from 'redux';
import auth from './Auth';
import articleListing from './articleListing';
import resetPassword from './resetPassword';
import changePassword from './changePassword';
import articles from './articles';
import editProfile from './editProfile';
import viewProfile from './viewProfile';
import membership from './Membership';
import viewArticle from './viewArticle';
import relatedArticles from './relatedArticles';
import bookmarks from './bookmarks';
import bookmarkArticle from './bookmarkArticle';
import comments from './Comments';
import search from './search';
import notifications from './notifications';
import rate from './rateArticle';

const reducers = combineReducers({
  auth,
  resetPassword,
  changePassword,
  articles,
  editProfile,
  viewProfile,
  membership,
  articleListing,
  viewArticle,
  relatedArticles,
  bookmarks,
  bookmarkArticle,
  comments,
  search,
  notifications,
  rate
});

export default reducers;
