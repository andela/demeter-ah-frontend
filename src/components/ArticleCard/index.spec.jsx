import React from 'react';
import { shallow, mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import ArticleCard from './index';
import store from '../../store';

describe('Sign In Component', () => {
  it('Should render without errors', () => {
    const component = shallow(
      <Router>
        <ArticleCard store={store} />
      </Router>
    );
    expect(component).toMatchSnapshot();
  });

  it('Should render along with children componenet', () => {
    const component = mount(
      <Router>
        <ArticleCard store={store} />
      </Router>
    );
    expect(component).toMatchSnapshot();
  });
});
