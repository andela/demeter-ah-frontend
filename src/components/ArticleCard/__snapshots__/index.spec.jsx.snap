// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign In Component Should render along with children componenet 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <ArticleCard
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <div
        className="md:flex md:flex-row flex-row w-full max-w-5xl m-auto text-left justify-start p-2 bg-white rounded-sm border-b-2 border-gray-350 mb-3 md:min-w-84"
      >
        <div
          className="m-4 min-w-full ml-0 min-w-108 min-h-46 rounded shadow-lg md:min-w-min lg:min-w-84"
          style={
            Object {
              "backgroundImage": "linear-gradient(180deg, rgba(0, 0, 0, 0.5) 99.98%, rgba(255, 255, 255, 0) 99.99%, rgba(255, 255, 255, 0.2) 100%), url(/placeholder.png)",
              "backgroundPosition": "center center",
              "backgroundRepeat": "no-repeat",
              "backgroundSize": "cover",
            }
          }
        >
          <div
            className="m-5 flex flex-col justify-between h-82"
          >
            <p
              className="text-xl font-medium text-white lg:mt-0 mt-4"
            />
            <div
              className="flex justify-between mt-16"
            >
              <div>
                <Link
                  to="/profile/Paul Igabu/articles"
                >
                  <a
                    href="/profile/Paul Igabu/articles"
                    onClick={[Function]}
                  >
                    <div
                      className="h-16 w-16"
                      style={
                        Object {
                          "backgroundImage": "url(/dummy.png)",
                          "backgroundPosition": "center center",
                          "backgroundSize": "cover",
                          "borderRadius": "9999px",
                        }
                      }
                    />
                    <p
                      className="text-base text-left font-medium text-white"
                    >
                      Paul Igabu
                    </p>
                  </a>
                </Link>
              </div>
              <span
                className="h-8"
                style={
                  Object {
                    "alignSelf": "flex-end",
                    "cursor": "pointer",
                    "width": "fit-content",
                  }
                }
              />
            </div>
          </div>
        </div>
        <div
          className="ml-0 md:ml-8 flex-grow justify-between flex flex-col max-h-48"
        >
          <div
            className="flex flex-col"
          >
            <Link
              to="/articles/undefined"
            >
              <a
                href="/articles/undefined"
                onClick={[Function]}
              >
                <h2
                  className="text-1.5xl lg:text-2xl capitalize"
                >
                  Title of this amazing article
                </h2>
              </a>
            </Link>
            <p
              className="text-xs"
            >
              Lorem ipsum dolor sit amet consectetur adipisicing elit. mus nihil eum iste! Dolorum, voluptatum officiis.
            </p>
          </div>
          <div
            className="flex justify-between mt-5"
          >
            <div
              className="flex w-2/3 justify-between text-sm"
            >
              <p
                className="pr-10 text-sm xl:text-base"
              >
                6 min
              </p>
              <p
                className="font-semibold text-sm xl:text-base"
              >
                July 12 2019
              </p>
            </div>
            <div
              className="flex w-1/3 justify-end items-center"
            >
              <Stars>
                <Star
                  key="0"
                >
                  <svg
                    fill="none"
                    height="17"
                    viewBox="0 0 13 12"
                    width="18"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M6.67928 0L8.09837 4.49139H12.6907L8.97542 7.26722L10.3945 11.7586L6.67928 8.98278L2.96403 11.7586L4.38313 7.26722L0.66788 4.49139H5.26018L6.67928 0Z"
                      fill="white"
                    />
                    <path
                      d="M6.67928 1.65959L7.6216 4.64203L7.73199 4.99139H8.09837H11.1861L8.67616 6.86667L8.39166 7.07923L8.49866 7.41786L9.44998 10.4288L6.97854 8.58223L6.67928 8.35863L6.38001 8.58223L3.90857 10.4288L4.85989 7.41786L4.96689 7.07923L4.68239 6.86667L2.17246 4.99139H5.26018H5.62656L5.73695 4.64203L6.67928 1.65959Z"
                      stroke="#C4C4C4"
                      strokeOpacity="0.6"
                    />
                  </svg>
                </Star>
                <Star
                  key="1"
                >
                  <svg
                    fill="none"
                    height="17"
                    viewBox="0 0 13 12"
                    width="18"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M6.67928 0L8.09837 4.49139H12.6907L8.97542 7.26722L10.3945 11.7586L6.67928 8.98278L2.96403 11.7586L4.38313 7.26722L0.66788 4.49139H5.26018L6.67928 0Z"
                      fill="white"
                    />
                    <path
                      d="M6.67928 1.65959L7.6216 4.64203L7.73199 4.99139H8.09837H11.1861L8.67616 6.86667L8.39166 7.07923L8.49866 7.41786L9.44998 10.4288L6.97854 8.58223L6.67928 8.35863L6.38001 8.58223L3.90857 10.4288L4.85989 7.41786L4.96689 7.07923L4.68239 6.86667L2.17246 4.99139H5.26018H5.62656L5.73695 4.64203L6.67928 1.65959Z"
                      stroke="#C4C4C4"
                      strokeOpacity="0.6"
                    />
                  </svg>
                </Star>
                <Star
                  key="2"
                >
                  <svg
                    fill="none"
                    height="17"
                    viewBox="0 0 13 12"
                    width="18"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M6.67928 0L8.09837 4.49139H12.6907L8.97542 7.26722L10.3945 11.7586L6.67928 8.98278L2.96403 11.7586L4.38313 7.26722L0.66788 4.49139H5.26018L6.67928 0Z"
                      fill="white"
                    />
                    <path
                      d="M6.67928 1.65959L7.6216 4.64203L7.73199 4.99139H8.09837H11.1861L8.67616 6.86667L8.39166 7.07923L8.49866 7.41786L9.44998 10.4288L6.97854 8.58223L6.67928 8.35863L6.38001 8.58223L3.90857 10.4288L4.85989 7.41786L4.96689 7.07923L4.68239 6.86667L2.17246 4.99139H5.26018H5.62656L5.73695 4.64203L6.67928 1.65959Z"
                      stroke="#C4C4C4"
                      strokeOpacity="0.6"
                    />
                  </svg>
                </Star>
                <Star
                  key="3"
                >
                  <svg
                    fill="none"
                    height="17"
                    viewBox="0 0 13 12"
                    width="18"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M6.67928 0L8.09837 4.49139H12.6907L8.97542 7.26722L10.3945 11.7586L6.67928 8.98278L2.96403 11.7586L4.38313 7.26722L0.66788 4.49139H5.26018L6.67928 0Z"
                      fill="white"
                    />
                    <path
                      d="M6.67928 1.65959L7.6216 4.64203L7.73199 4.99139H8.09837H11.1861L8.67616 6.86667L8.39166 7.07923L8.49866 7.41786L9.44998 10.4288L6.97854 8.58223L6.67928 8.35863L6.38001 8.58223L3.90857 10.4288L4.85989 7.41786L4.96689 7.07923L4.68239 6.86667L2.17246 4.99139H5.26018H5.62656L5.73695 4.64203L6.67928 1.65959Z"
                      stroke="#C4C4C4"
                      strokeOpacity="0.6"
                    />
                  </svg>
                </Star>
                <Star
                  key="4"
                >
                  <svg
                    fill="none"
                    height="17"
                    viewBox="0 0 13 12"
                    width="18"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M6.67928 0L8.09837 4.49139H12.6907L8.97542 7.26722L10.3945 11.7586L6.67928 8.98278L2.96403 11.7586L4.38313 7.26722L0.66788 4.49139H5.26018L6.67928 0Z"
                      fill="white"
                    />
                    <path
                      d="M6.67928 1.65959L7.6216 4.64203L7.73199 4.99139H8.09837H11.1861L8.67616 6.86667L8.39166 7.07923L8.49866 7.41786L9.44998 10.4288L6.97854 8.58223L6.67928 8.35863L6.38001 8.58223L3.90857 10.4288L4.85989 7.41786L4.96689 7.07923L4.68239 6.86667L2.17246 4.99139H5.26018H5.62656L5.73695 4.64203L6.67928 1.65959Z"
                      stroke="#C4C4C4"
                      strokeOpacity="0.6"
                    />
                  </svg>
                </Star>
              </Stars>
            </div>
          </div>
        </div>
      </div>
    </ArticleCard>
  </Router>
</BrowserRouter>
`;

exports[`Sign In Component Should render without errors 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <ArticleCard
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  />
</Router>
`;
